darwin/cli_functions.py

```diff
@@ -428,8 +429,14 @@ def pull_dataset(
         Pulls video frames images instead of video files. Defaults to False.
     force_slots: bool
         Pulls all slots of items into deeper file structure ({prefix}/{item_name}/{slot_name}/{file_name})
+    retry: bool
+        If True, will repeatedly try to download the release if it is still processing up to a maximum of 5 minutes.
     """
     version: str = DatasetIdentifier.parse(dataset_slug).version or "latest"
+    if version == "latest" and retry:
```

Comment:
Few questions here:
- I don't recall the details but is the name `latest` hardcoded by us? 
- What happens if a client deliberately passes the name `latest` with `retry=True`? 
- I don't think this restriction is necessary, can't we pick the name of the latest release ourselves before performing the download and then do the retry logic using it instead of `latest`. This would ensure we refer to the same export even if a new export would be created in the meantime

Reply:
> I don't recall the details but is the name latest hardcoded by us?

Yes, `latest` is a reserved release name. If you try to create an export named `latest`, the api responds with `{"errors":{"name":["is reserved"]}}`

> What happens if a client deliberately passes the name `latest` with `retry=True`?

[We will return the latest available release](https://github.com/v7labs/darwin-py/blob/DAR-2707/darwin/dataset/remote_dataset.py#L799). Since it is never possible to create an export with the name `latest`, should we disallow this when using `get_release()` to prevent confusion?

>I don't think this restriction is necessary, can't we pick the name of the latest release ourselves before performing the download and then do the retry logic using it instead of latest. This would ensure we refer to the same export even if a new export would be created in the meantime

Actually yes, I think we can. This is because each release has a `export_date` of type `datetime.datetime`. This allows us to select the most recent release incase `retry` is passed as `True`. I'll make this change now